pages/bookshelf.tsx

```diff
@@ -0,0 +1,53 @@
+import { NextPage } from "next"
+import "react-toastify/dist/ReactToastify.css"
+import Navigation from "../components/bookshelf/Navigation"
+import { Container } from "../components/styled/bookshelf"
+import { useState, useContext, useEffect, useMemo } from "react"
+import Discover from "../components/bookshelf/discover/Discover"
+import LogoutBox from "../components/bookshelf/LogoutBox"
+import { UserContext } from "../context/UserContext"
+import { useRouter } from "next/router"
+import { getBooks } from "../api/books"
+import { BooksContext } from "../context/BooksContext"
+
+const Home: NextPage = () => {
+  const [active, setActive] = useState("Reading List")
+  const { user, setUser } = useContext(UserContext)
+  const { books, setBooks } = useContext(BooksContext)
+  const value = useMemo(() => ({ books, setBooks }), [books, setBooks])
+  const router = useRouter()
+
+  useEffect(() => {
+    const existingUser = JSON.parse(localStorage.getItem("user"))
+    existingUser ? setUser(existingUser) : router.push("/")
+
+    const existingBooks = JSON.parse(localStorage.getItem("books"))
+    existingBooks ? setBooks(existingBooks) : ""
```

Comment:
if nothing's happening on the false_conditition, I think this should work out

`existingBooks && setBooks(existingBooks)`, if not
`if(existingBooks) setBooks(existingBooks)`

there's an expectation that there's something that will happen at the false side of the ternary operator with the current code.

Reply:
fixed `if (existingBooks) setBooks(existingBooks)`